Pause quality monitors when "mode: development" is used

lennartkats-db opened the pull request:
## Changes

Similar to scheduled jobs, quality monitors should be paused when in development mode (in line with the [behavior for scheduled jobs](https://docs.databricks.com/en/dev-tools/bundles/deployment-modes.html)). @aravind-segu @arpitjasa-db please take a look and verify this behavior.

- [x] Followup: documentation changes. If we make this change we should update https://docs.databricks.com/dev-tools/bundles/deployment-modes.html.

## Tests
Unit tests

arpitjasa-db commented:
@vperiyasamy to confirm behavior from Monitoring side

lennartkats-db commented:
Related PR, generalizing the "mode: development" behavior: https://github.com/databricks/cli/pull/1490

vperiyasamy reviewed (changes requested):
We actually don't support pausing the monitor schedules directly, the schedule status is a read-only field that we exposed because we were considering building an "auto pausing" feature which would detect successive failures. Is it required for us to have this behavior? If so, we can delete the schedule to "pause" it.

lennartkats-db commented:
@aravind-segu @arpitjasa-db  @vperiyasamy I updated the PR to remove the schedule instead for now, please take another look. 

I also updated the config test so it actually works out of the box on staging in order to verify this change.

vperiyasamy commented on bundle/config/mutator/process_target_mode.go:
```diff
@@ -105,6 +106,15 @@ func transformDevelopmentMode(ctx context.Context, b *bundle.Bundle) diag.Diagno
 		// (registered models in Unity Catalog don't yet support tags)
 	}
 
+	for i := range r.QualityMonitors {
```
just to confirm: this does not mess with the production version of these monitors, which should still have schedules intact?

lennartkats-db replied:
Yeah, indeed, `transformDevelopmentMode()` is only applied when a deployment target uses `mode: development`.

vperiyasamy commented on bundle/tests/quality_monitor/databricks.yml:
```diff
@@ -24,14 +31,14 @@ targets:
   production:
     resources:
       quality_monitors:
         my_monitor:
```
what does this name mean?

lennartkats-db replied:
The original name wasn't valid according the the API. It requires something of the format `catalog.schema`. Actually, that seems like something to reflect on https://docs.databricks.com/api/workspace/qualitymonitors/create?

pietern commented on bundle/tests/quality_monitor/databricks.yml:
```diff
@@ -24,14 +31,14 @@ targets:
         my_monitor:
           table_name: "main.test.staging"
-          output_schema_name: "staging"
+          output_schema_name: "main.test"
```
You can omit these overrides since they no longer override anything.

lennartkats-db replied:
Let me just change these values again to show how to differentiate them a bit.

pietern reviewed (approved):
Minor comment about the test configuration. Rest LGTM.

arpitjasa-db commented on bundle/tests/quality_monitor/databricks.yml:
```diff
@@ -1,19 +1,26 @@
   quality_monitors:
     my_monitor:
-      table_name: "main.test.thing1"
+      table_name: "numbers"
```
Is there a reason we don't need to provide `catalog.schema.table` here and can only provide table?

lennartkats-db replied:
Ah, let me just change that back. Note that `table_name` is actually overridden in all targets in the existing example.
